asserts/database.go

```diff
@@ -408,6 +427,25 @@ func (db *Database) Add(assert Assertion) error {
 		return fmt.Errorf("cannot add %q assertion with primary key clashing with a predefined assertion: %v", ref.Type.Name, ref.PrimaryKey)
 	}
 
+	// this is non empty only in the stacked case
+	stackedOn := db.backstores[3:]
```

Comment:
Wait, why 3?

I would feel much more comfortable with a helper, like `db.stackedOn()` returning that. It feels a little bit too magic.